src/ripple/protocol/impl/IOUAmount.cpp

```diff
@@ -0,0 +1,147 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/protocol/IOUAmount.h>
+
+namespace ripple {
+
+void
+IOUAmount::normalize ()
+{
+    /* The range for the exponent when normalized */
+    static int const minExponent = -96;
+    static int const maxExponent = 80;
+
+    /* The range for the mantissa when normalized */
+    static std::int64_t const minMantissa = 1000000000000000ull;
+    static std::int64_t const maxMantissa = 9999999999999999ull;
+
+    if (mantissa_ == 0)
+    {
+        *this = zero;
+        return;
+    }
+
+    bool const negative = (mantissa_ < 0);
```

Comment:
"It depends."  Branches can be very expensive because you can lose your instruction pipeline if branch prediction fails.